lib/alchemy/page_layout.rb

```diff
@@ -127,7 +133,28 @@ def human_layout_name(layout)
       # Returns true if the given layout is unique and not already taken or it should be hidden.
       #
       def layout_available?(layout)
-        !layout['hide'] && !already_taken?(layout) && available_on_site?(layout)
+        layout_available_from_parent?(layout) && layout_available_from_child?(layout) &&
+          !already_taken?(layout) && available_on_site?(layout)
+      end
+
+      def layout_available_from_parent?(layout)
+        return true if !@page_layout
+
+        includes = @page_layout.dig('child_pages', 'include')
+        excludes = @page_layout.dig('child_pages', 'exclude')
+
+        @page_layout['child_pages'].blank? ||
+          includes &.include?(layout['name']) ||
+          (excludes && !excludes.include?(layout['name']))
+      end
+
+      def layout_available_from_child?(layout)
```

Comment:
It's not about creating a parent for a child. But under which parents the layout will be available. Previously there was a setting `hide = true` which made certain layouts hidden from creation. I just deleted this `hide` setting from the conditions in my last push as this can be achived with `layout['parent_pages'] = false.

But the reason why I think this feature makes sense to have is the following. Imagine that you have a mature website with some 15 page layouts available. Then you want a new type of blog post layout `special_blog_post` that should only be creatable under the blog layout.

With the new feature you simply set `parent_pages['only'] = ['blog']` in the `special_blog_post` settings.

Without it you would have to revisit all other page_layout and append `child_pages['except'] = [ ... 'special_blog_post']  

... Which feels a lot more tedious/error prone. In the end this logic is only executed on admin page creation, and we're already looping through all layouts to check the rest of the conditions - so i dont think it could have any significant impact on performance?

What do you think?